WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProductVariationToOrder.swift

```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+
+/// View showing a list of product variations to add to an order.
+///
+struct AddProductVariationToOrder: View {
+    /// Defines whether the view is presented.
+    ///
+    @Binding var isPresented: Bool
+
+    /// View model to drive the view.
+    ///
+    @ObservedObject var viewModel: AddProductVariationToOrderViewModel
+
+    var body: some View {
+        Group {
+            switch viewModel.syncStatus {
+            case .results:
+                InfiniteScrollList(isLoading: viewModel.shouldShowScrollIndicator,
+                                   loadAction: viewModel.syncNextPage) {
+                    ForEach(viewModel.productVariationRows) { rowViewModel in
+                        ProductRow(viewModel: rowViewModel)
+                            .onTapGesture {
+                                viewModel.selectVariation(rowViewModel.productOrVariationID)
+                                isPresented.toggle()
+                            }
+                    }
+                }
+            case .empty:
+                EmptyState(title: Localization.emptyStateMessage, image: .emptyProductsTabImage)
+                    .frame(maxHeight: .infinity)
+            case .firstPageSync:
+                List(viewModel.ghostRows) { rowViewModel in
+                    ProductRow(viewModel: rowViewModel)
+                        .redacted(reason: .placeholder)
+                        .shimmering()
+                }
+                .listStyle(PlainListStyle())
+            default:
+                EmptyView()
+            }
+        }
+        .background(Color(.listBackground).ignoresSafeArea())
+        .ignoresSafeArea(.container, edges: .horizontal)
+        .navigationTitle(viewModel.productName)
+        .navigationBarTitleDisplayMode(.inline)
+        .onAppear {
+            viewModel.syncFirstPage()
```

Comment:
Do you think we will have problems with this happening on `onAppear`?
For example, if a system alert happens or if the user navigates to other apps and then comes back, this would be re-invoked and will possibly corrupt the sync state.

A safer approach could be to apply the following pattern:

- Have the view model to expose a `onLoadTrigger` subject

https://github.com/woocommerce/woocommerce-ios/blob/4f661ab9d0fc101b1755ef482c6d77afa827fcd2/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Address%20Edit/AddressFormViewModelProtocol.swift#L212-L214

- Call that trigger on `onAppear`

https://github.com/woocommerce/woocommerce-ios/blob/2576e3f8b7377175454fd359367bd1961eaca4b1/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Address%20Edit/EditOrderAddressForm.swift#L267-L269

- Subscribe to that publisher but only listen for the first event.
https://github.com/woocommerce/woocommerce-ios/blob/4f661ab9d0fc101b1755ef482c6d77afa827fcd2/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Address%20Edit/AddressFormViewModelProtocol.swift#L174-L186

What do you think?